packages/storage-backend/storage-lib/src/main/java/com/daimler/data/minio/client/DnaMinioClientImp.java

```diff
@@ -842,13 +841,19 @@ public MinioGenericResponse removeObjects(String userId, String bucketName, Stri
 					for (String path: prefixValue) {
 					    for (String objectName: objectNames) {
 							 if (objectName.contains(path)) {
-								objects.add(new DeleteObject(objectName));
+								 // Fetch the versions for this object
+								 Iterable<Result<Item>> versions = minioClient.listObjects(ListObjectsArgs.builder().bucket(bucketName).prefix(objectName).includeVersions(true).build());
+								 for (Result<Item> versionResult : versions) {
+									 Item versionItem = versionResult.get();
+									 // Add both the object name and version ID to delete this specific version.
+									 objects.add(new DeleteObject(objectName, versionItem.versionId()));
+								 }
 							 }
 					    }
                     }
                 }
 
-                LOGGER.info("Removing objects from Minio bucket:{}", bucketName);
+				LOGGER.info("Removing objects from Minio bucket:{}", bucketName);
 				Iterable<Result<DeleteError>> results = minioClient
 						.removeObjects(RemoveObjectsArgs.builder().bucket(bucketName).objects(objects).build());
```

Comment:
.objects(objects).bypassGovernanceMode(true)

if this works, then only remove above code